call-notes/Epoch Rewards.md

```diff
@@ -0,0 +1,171 @@
+
+### Epoch Reward History Sysvar Account  
+Video | [5:31](https://youtu.be/IbviAInuSHk?list=PLilwLeBwGuK7e_mH_sFwTytYQxalh7xd5&t=331)
+ -|-
+
+With the new approach the rewards are not going to be distributed at one particular block instead is going to be spread out between multiple blocks. In order to track the progress of a rewarded distribution there will be two system sysvar of our account that are added to help to track the rewarded distribution the first one is the epoch rewarded history Sysvar. It is basically a 500 fixed-size array to be consistent with the stake history we choose that number to be 512 and it's a fixed-size array and each entry in the array contains three elements, three fields the first one is the total rewarding land port for the epoch. The second one is how much reward has already been distributed in import that's the progress and the other one the last one is the hash of all the rewards that are going to be paid out. This basically is introduced to verify the rewarded distribution. It's similar to the role of account hash but it's just more specific for rewards it's only harsh from all the rewards that are going to be paid out.
+
+### Epoch Reward Reserve Sysvar Account 
+
+Video | [6:55](https://youtu.be/IbviAInuSHk?list=PLilwLeBwGuK7e_mH_sFwTytYQxalh7xd5&t=416)
+ -|-
+
+The second is that the account to introduce is called Epoch reward reserve. As I mentioned earlier in reward distribution, each block of all the rewards are going to be distributed in M blocks so we introduced the second set of sysvars this is basically M sysvars that keeps tracks of all the rewards that are going to be distributed in one block and similarly it has a balance field in which it describes how much rewards are going to be distributed for this particular block and it also has a hash that's the hash of all the rewards are going to be distributed in this block and since we have a M sysvar account, so the address of the of the one particular system account for a particular block is basically determined by harsh, the base ID to the block height then we will get the unique address for the rewards going to be distributed.
+
+### Reward Hash  
+
+Video | [7:58](https://youtu.be/IbviAInuSHk?list=PLilwLeBwGuK7e_mH_sFwTytYQxalh7xd5&t=478)
+ -|-
+
+The relationship between the hash and the root hash in the Epoch history sysvar is the root hash as we seen earlier here, this is the root hash and it's computed by accumulating all the hashes from all the reserves together. So after we credit all the reserves, credit order rewards from the reserves we will compare the hash against the rule hash to make sure they match each other so that we can be sure that the reward distribution is correct. Okay,  that's the main changes of the proposal.

```

Comment:
```suggestion
The relationship between the hash and the root hash in the Epoch history sysvar is the root hash as we seen earlier here, this is the root hash and it's computed by accumulating all the hashes from all the reserves together. So after we credit all the reserves, credit order rewards from the reserves we will compare the hash against the root hash to make sure they match each other so that we can be sure that the reward distribution is correct. Okay,  that's the main changes of the proposal.

```